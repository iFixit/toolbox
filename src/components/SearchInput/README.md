### Examples

#### Standard SearchInput

```
<SearchInput placeholder="Search" />
```

#### SearchInput with value

```
<SearchInput value="iphone 6" />
```

#### Disabled SearchInput

```
<SearchInput value="toolkit" disabled />
```

#### Event callbacks

```
/* Hint: Check the console */
<SearchInput
  placeholder="Name"
  onChange={event => console.log(event.target.value)}
  onFocus={() => console.log('focused')}
  onBlur={() => console.log('blurred')}
/>
```
