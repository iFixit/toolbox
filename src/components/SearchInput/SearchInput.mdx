---
name: SearchInput
route: /components/search-input
menu: Components
---

import SearchInput from './SearchInput'
import { Playground, PropsTable } from 'docz'

# SearchInput

## Examples

### Default

<Playground>
  <SearchInput placeholder="Search" />
</Playground>

### SearchInput with value

<Playground>
  <SearchInput value="iphone 6" />
</Playground>

### Disabled SearchInput

<Playground>
  <SearchInput value="toolkit" disabled />
</Playground>

### Event callbacks

<Playground>
  {/* Hint: Check the console */}
  <SearchInput
    placeholder="Search"
    onChange={event => console.log(event.target.value)}
    onFocus={() => console.log('focused')}
    onBlur={() => console.log('blurred')}
  />
</Playground>


## Props

<PropsTable of={SearchInput} />
