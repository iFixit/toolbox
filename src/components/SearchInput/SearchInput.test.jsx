import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import SearchInput from './SearchInput';

test('renders without crashing', () => {
   const tree = renderer.create(<SearchInput />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders when disabled', () => {
   const tree = renderer.create(<SearchInput disabled />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('calls onChange when input value changes', () => {
   const handleChange = jest.fn();
   const wrapper = mount(<SearchInput onChange={handleChange} />);
   wrapper.find('input').simulate('change', { target: { value: 'test' } });
   expect(handleChange).toHaveBeenCalledTimes(1);
});
