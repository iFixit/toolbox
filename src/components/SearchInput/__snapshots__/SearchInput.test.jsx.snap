// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when disabled 1`] = `
.glamor-3,
[data-glamor-3] {
  position: relative;
}

.glamor-1,
[data-glamor-1] {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  pointer-events: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-0,
[data-glamor-0] {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: #9E9FA0;
  stroke-width: 2;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: content-box;
}

.glamor-2,
[data-glamor-2] {
  -webkit-appearance: none;
  box-sizing: border-box;
  display: block;
  width: 100%;
  margin: 0;
  padding: 0.75rem 1rem 0.75rem 56px;
  font-family: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: rgba(0, 3, 6, 0.38);
  background-color: rgba(0, 3, 6, 0.04);
  border: none;
  border-radius: 4px;
  outline: none;
  box-shadow: inset 0 0 0 1px rgba(0, 3, 6, 0.12);
  transition: box-shadow 150ms ease-in-out;
  pointer-events: none;
  -webkit-transition: box-shadow 150ms ease-in-out;
  -moz-transition: box-shadow 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover,
.glamor-2 :focus,
[data-glamor-2] :focus {
  box-shadow: 0 0 0 1px rgba(0, 3, 6, 0.12), 0 4px 8px 0 rgba(0, 3, 6, 0.12), 0 2px 4px 0 rgba(0, 3, 6, 0.07);
}

.glamor-2::-webkit-search-decoration,
[data-glamor-2]::-webkit-search-decoration {
  display: none;
}

<div
  className="glamor-3"
>
  <div
    className="glamor-1"
  >
    <span
      className="glamor-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-search\\"><circle cx=\\"10.5\\" cy=\\"10.5\\" r=\\"7.5\\"></circle><line x1=\\"21\\" y1=\\"21\\" x2=\\"15.8\\" y2=\\"15.8\\"></line></svg>",
        }
      }
    />
  </div>
  <input
    className="glamor-2"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    required={false}
    type="search"
    value=""
  />
</div>
`;

exports[`renders without crashing 1`] = `
.glamor-3,
[data-glamor-3] {
  position: relative;
}

.glamor-1,
[data-glamor-1] {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  align-items: center;
  pointer-events: none;
  -webkit-box-align: center;
  -webkit-align-items: center;
}

.glamor-0,
[data-glamor-0] {
  display: inline-block;
  width: 24px;
  height: 24px;
  color: #9E9FA0;
  stroke-width: 2;
  padding-left: 1rem;
  padding-right: 1rem;
  box-sizing: content-box;
}

.glamor-2,
[data-glamor-2] {
  -webkit-appearance: none;
  box-sizing: border-box;
  display: block;
  width: 100%;
  margin: 0;
  padding: 0.75rem 1rem 0.75rem 56px;
  font-family: inherit;
  font-size: 1rem;
  line-height: 1.5;
  color: rgba(0, 3, 6, 0.87);
  background-color: #FFFFFF;
  border: none;
  border-radius: 4px;
  outline: none;
  box-shadow: 0 0 0 1px rgba(0, 3, 6, 0.12), 0 2px 4px 0 rgba(0, 3, 6, 0.07);
  transition: box-shadow 150ms ease-in-out;
  -webkit-transition: box-shadow 150ms ease-in-out;
  -moz-transition: box-shadow 150ms ease-in-out;
}

.glamor-2:hover,
[data-glamor-2]:hover,
.glamor-2 :focus,
[data-glamor-2] :focus {
  box-shadow: 0 0 0 1px rgba(0, 3, 6, 0.12), 0 4px 8px 0 rgba(0, 3, 6, 0.12), 0 2px 4px 0 rgba(0, 3, 6, 0.07);
}

.glamor-2::-webkit-search-decoration,
[data-glamor-2]::-webkit-search-decoration {
  display: none;
}

<div
  className="glamor-3"
>
  <div
    className="glamor-1"
  >
    <span
      className="glamor-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"100%\\" height=\\"100%\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"inherit\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" class=\\"feather feather-search\\"><circle cx=\\"10.5\\" cy=\\"10.5\\" r=\\"7.5\\"></circle><line x1=\\"21\\" y1=\\"21\\" x2=\\"15.8\\" y2=\\"15.8\\"></line></svg>",
        }
      }
    />
  </div>
  <input
    className="glamor-2"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Search"
    required={false}
    type="search"
    value=""
  />
</div>
`;
